arangod/RestServer/FlushFeature.cpp

```diff
@@ -578,119 +74,22 @@ void FlushFeature::collectOptions(std::shared_ptr<ProgramOptions> options) {
-    auto* logFileManager = MMFilesLogfileManager::instance(true); // true to avoid assertion failure
-
-    if (!logFileManager) {
-      LOG_TOPIC("038b3", ERR, Logger::FLUSH)
-        << "failed to find an MMFiles log file manager instance while registering "
-           "'Flush' marker subscription for type '" << type << "'";
-
-      return nullptr;
-    }
-
-    subscription = std::make_shared<MMFilesFlushSubscription>(type, vocbase.id(), *logFileManager);
-  } else if (EngineSelectorFeature::isRocksDB()) {
-#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
-    auto* rocksdbEngine = dynamic_cast<RocksDBEngine*>(engine);
-#else
-    auto* rocksdbEngine = static_cast<RocksDBEngine*>(engine);
-#endif
-    TRI_ASSERT(rocksdbEngine);
-
-    auto* db = rocksdbEngine->db();
-
-    if (!db) {
-      LOG_TOPIC("0f0f6", ERR, Logger::FLUSH)
-       << "failed to find a RocksDB engine db while registering "
-          "'Flush' marker subscription for type '" << type << "'";
-
-      return nullptr;
-    }
-
-    auto* rootDb = db->GetRootDB();
-
-    if (!rootDb) {
-      LOG_TOPIC("26c23", ERR, Logger::FLUSH)
-        << "failed to find a RocksDB engine root db while registering "
-           "'Flush' marker subscription for type '" << type << "'";
-
-      return nullptr;
-    }
-
-    subscription = std::make_shared<RocksDBFlushSubscription>(type, vocbase.id(), *rootDb);
-  }
-  #ifdef ARANGODB_USE_GOOGLE_TESTS
-    else if (_defaultFlushSubscription) {
-      struct DelegatingFlushSubscription final: public FlushSubscriptionBase {
-        DelegatingFlushSubscription(
-            std::string const& type,
-            TRI_vocbase_t const& vocbase,
-            DefaultFlushSubscription& delegate)
-          : FlushSubscriptionBase(type, vocbase.id()),
-           _delegate(delegate),
-           _vocbase(vocbase) {
-        }
-
-        Result commitImpl(VPackSlice data, TRI_voc_tick_t tick) override {
-          auto const res = _delegate(_type, _vocbase, data, tick);
-
-          if (res.ok()) {
-            resetCurrentTick(tick);
-          }
-
-          return res;
-        }
-
-        DefaultFlushSubscription _delegate;
-        TRI_vocbase_t const& _vocbase;
-      };
-
-      subscription = std::make_shared<DelegatingFlushSubscription>(type, vocbase, _defaultFlushSubscription);
-    }
-  #endif
-
+void FlushFeature::registerFlushSubscription(const std::shared_ptr<FlushSubscription>& subscription) {
   if (!subscription) {
-    LOG_TOPIC("53c4e", ERR, Logger::FLUSH)
-      << "failed to identify storage engine while registering "
-         "'Flush' marker subscription for type '" << type << "'";
-
-    return nullptr;
+    return;
   }
 
   std::lock_guard<std::mutex> lock(_flushSubscriptionsMutex);
 
   if (_stopped) {
     LOG_TOPIC("798c4", ERR, Logger::FLUSH) << "FlushFeature not running";
 
-    return nullptr;
+    return;
   }
 
   _flushSubscriptions.emplace_back(subscription);
 
-  return subscription;
+  return;
```

Comment:
this return statement seems unnecessary now
```suggestion
```

Reply:
done